pallets/did/src/did_details.rs

```diff
@@ -670,37 +670,37 @@ pub trait DeriveDidCallAuthorizationVerificationKeyRelationship {
 /// extrinsic to have a DID origin and perform DID-based authorization upon
 /// their invocation.
 #[derive(Clone, RuntimeDebug, Decode, Encode, PartialEq, TypeInfo)]
-#[scale_info(skip_type_params(T))]
 
-pub struct DidAuthorizedCallOperation<T: Config> {
+pub struct DidAuthorizedCallOperation<DidIdentifier, DidCallable, BlockNumber, AccountId> {
 	/// The DID identifier.
-	pub did: DidIdentifierOf<T>,
+	pub did: DidIdentifier,
 	/// The DID tx counter.
 	pub tx_counter: u64,
```

Comment:
I think also the tx counter could be generic.

Reply:
And required to implement the `CheckedAdd` or `WrappingAdd` only for the function(s) that change its value.